### PR TITLE
OBPIH-4341 Add option to add two-way associations

### PR DESCRIPTION
**Pull request by awalkowiak**

### CODE REVIEW

File: grails-app/controllers/org/pih/warehouse/product/ProductAssociationController.groovy [hunk 1]
```diff
@@ -73,6 +73,15 @@ class ProductAssociationController {
...
         def productAssociationInstance = new ProductAssociation(params)
         if (productAssociationInstance.save(flush: true)) {
+            if (params.hasMutualAssociation) {
+                def mutualAssociationInstance = new ProductAssociation()
+                bindMutualAssociationData(mutualAssociationInstance, params)
+
+                mutualAssociationInstance.mutualAssociation = productAssociationInstance
+                productAssociationInstance.mutualAssociation = mutualAssociationInstance
+
+                mutualAssociationInstance.save(flush: true, failOnError: true)
```
Comment (jmiranda) [inline grails-app/controllers/org/pih/warehouse/product/ProductAssociationController.groovy:83]: Again we should add something to undo these changes if the user unchecks the checkbox. Ask Kelsey what she thinks.

File: grails-app/views/productAssociation/create.gsp [hunk 2]
```diff
@@ -75,6 +75,15 @@
 									<g:textArea class="text" name="comments" value="${productAssociationInstance?.comments}" />
 								</td>
 							</tr>
+
+							<tr class="prop">
+								<td class="name">
+									<label><warehouse:message code="productAssociation.mutualAssociation.label" default="Two-way Association"/></label>
```
Comment (jmiranda) [inline grails-app/views/productAssociation/create.gsp:81]: "Two-way association" needs some work. Bi-directional is the only thing I can think of and that's not any better.

**Review by jmiranda — APPROVED** [review]

File: grails-app/views/productAssociation/create.gsp (hunk 2 above)
Comment (jmiranda) [reply grails-app/views/productAssociation/create.gsp:81]: Ok sounds good. My vote would be for Bi-directional association or Symmetric association. Or even something more laymen like "Do you want to create an inverse relationship for the associated product?"

**Comment by awalkowiak** [comment]
@jmiranda I added delete handling for mutual associations according to Kelsey's request here: https://pihemr.atlassian.net/browse/OBPIH-4341?focusedCommentId=131002
